python/python/bystro/proteomics/fragpipe_tandem_mass_tag.py

```diff
@@ -15,6 +15,25 @@ class TandemMassTagDataset:
     abundance_df: pd.DataFrame
     annotation_df: pd.DataFrame
 
+    def __post_init__(self) -> None:
+        try:
+            check_df_starts_with_cols(self.abundance_df, ABUNDANCE_COLS[1:])
+        except ValueError as e:
+            err_msg = "Received abundance_df with unexpected columns"
+            raise ValueError(err_msg) from e
+
+    def get_melted_abundance_df(self) -> pd.DataFrame:
+        """Return a melted abundance df with columns [gene_name, sample_id, value]"""
+        abundance_df = self.abundance_df
+        columns_to_drop = ["NumberPSM", "ProteinID", "MaxPepProb", "ReferenceIntensity"]
```

Comment:
Are these columns always guaranteed? Also, isn't there a chance there are more extra columns than this? I think having extra columns should be allowed without throwing an error as long as we have the expected starting columns. In that case, it might be more worthwhile to focus on columns you want to keep and then subtract out everything you want to drop without specifying the extra column names?

Reply:
Yes, these columns are guaranteed by the `__post_init__` validator of this class.  In this method we're converting the df from wide to long format, i.e. converting a matrix with labeled rows and columns to a table of the form `(row_label, col_label, value)`, and that conversion might silently fail if there are additional unexpected columns present.  So I would opt for the strict validator that will fail on unexpected input data, and if it chokes on unexpected data later, the class can be revised in order to accommodate both input types.  Wdyt?

Reply:
I think that's a good strategy for now. In future, I'm worried there's a pretty high chance there will be additional columns in the input files, and I can imagine a point at which we might want to be able to keep the extra columns for some kind pre-processing purposes before analysis as we expand functionality.  I think we should be able to take in everything as long as the required columns are present and just drop the extra columns in terms of requirements for analysis. Though, for now, I can just keep this as a note for future work.